src/UI/Actions.hs

```diff
@@ -595,3 +601,14 @@ getMailsForThread
 getMailsForThread (_, ts) s =
   let dbpath = view (asConfig . confNotmuch . nmDatabase) s
   in either (const mempty)(view vector . itoList) <$> runExceptT (Notmuch.getThreadMessages dbpath ts)
+
+reloadThreadTags
+  :: MonadIO m
+  => AppState
+  -> (a, NotmuchThread)
+  -> m (AppState -> AppState)
+reloadThreadTags s (_, thread) =
+  let dbpath = view (asConfig. confNotmuch . nmDatabase) s
+      updateList t' = over (asMailIndex . miListOfThreads) (L.listModify $ const t')
+  in runExceptT (Notmuch.reloadTags dbpath thread)
+     >>= either (pure . setError) (pure . updateList)
```

Comment:
This looks like it could be simplified to:
```haskell
let
  ...
in
  either setError updateList <$> runExceptT (Notmuch.reloadTags dbpath thread)
```

Reply:
Is the point of using fmap to simplify that I operate in the Monad instance of the Functor? (Not sure if that sentence makes sense).

Reply:
yeah, using the weaker abstraction makes it clearer what's going on, i.e. that the next
action cannot depend on the result of the first.  It also avoids two redundant applications of `pure`.